### PR TITLE
Ignore warnings from packaging module when testing import behaviour.

### LINKED ISSUE #6831: test_no_accidental_warnings fails with packaging >= 20.5
numba 0.53 on openSUSE build service for Tumbleweed:

* Python 3.8.8
* packaging 20.8
* setuptools 44.1.1

```
[  745s] ======================================================================
[  745s] FAIL: test_no_accidental_warnings (numba.tests.test_import.TestNumbaImport)
[  745s] ----------------------------------------------------------------------
[  745s] Traceback (most recent call last):
[  745s]   File "/usr/lib64/python3.8/site-packages/numba/tests/test_import.py", line 62, in test_no_accidental_warnings
[  745s]     self.run_in_subproc(code, flags)
[  745s]   File "/usr/lib64/python3.8/site-packages/numba/tests/test_import.py", line 22, in run_in_subproc
[  745s]     raise AssertionError(msg % (popen.returncode, err.decode()))
[  745s] AssertionError: process failed with code 1: stderr follows
[  745s] Traceback (most recent call last):
[  745s]   File "/usr/lib/python3.8/site-packages/packaging/version.py", line 57, in parse
[  745s]     return Version(version)
[  745s]   File "/usr/lib/python3.8/site-packages/packaging/version.py", line 298, in __init__
[  745s]     raise InvalidVersion("Invalid version: '{0}'".format(version))
[  745s] packaging.version.InvalidVersion: Invalid version: 'numpy'
[  745s] 
[  745s] During handling of the above exception, another exception occurred:
[  745s] 
[  745s] Traceback (most recent call last):
[  745s]   File "<string>", line 1, in <module>
[  745s]   File "/usr/lib64/python3.8/site-packages/numba/__init__.py", line 19, in <module>
[  745s]     from numba.core import config
[  745s]   File "/usr/lib64/python3.8/site-packages/numba/core/config.py", line 16, in <module>
[  745s]     import llvmlite.binding as ll
[  745s]   File "/usr/lib64/python3.8/site-packages/llvmlite/binding/__init__.py", line 4, in <module>
[  745s]     from .dylib import *
[  745s]   File "/usr/lib64/python3.8/site-packages/llvmlite/binding/dylib.py", line 3, in <module>
[  745s]     from llvmlite.binding import ffi
[  745s]   File "/usr/lib64/python3.8/site-packages/llvmlite/binding/ffi.py", line 175, in <module>
[  745s]     from pkg_resources import resource_filename
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3252, in <module>
[  745s]     def _initialize_master_working_set():
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3235, in _call_aside
[  745s]     f(*args, **kwargs)
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3264, in _initialize_master_working_set
[  745s]     working_set = WorkingSet._build_master()
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 574, in _build_master
[  745s]     ws = cls()
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 567, in __init__
[  745s]     self.add_entry(entry)
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 623, in add_entry
[  745s]     for dist in find_distributions(entry, True):
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2061, in find_on_path
[  745s]     path_item_entries = _by_version_descending(filtered)
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2034, in _by_version_descending
[  745s]     return sorted(names, key=_by_version, reverse=True)
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2032, in _by_version
[  745s]     return [packaging.version.parse(part) for part in parts]
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2032, in <listcomp>
[  745s]     return [packaging.version.parse(part) for part in parts]
[  745s]   File "/usr/lib/python3.8/site-packages/packaging/version.py", line 59, in parse
[  745s]     return LegacyVersion(version)
[  745s]   File "/usr/lib/python3.8/site-packages/packaging/version.py", line 127, in __init__
[  745s]     warnings.warn(
[  745s] DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release

```

This is another fallout of https://github.com/pypa/setuptools/issues/2466

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

Fixes #6831

### CODE REVIEW

File: numba/tests/test_import.py [hunk 1]
```diff
@@ -58,7 +58,9 @@ def test_no_accidental_warnings(self):
         # checks that importing Numba isn't accidentally triggering warnings due
         # to e.g. deprecated use of import locations from Python's stdlib
         code = "import numba"
-        flags = ["-Werror",]
+        # See: https://github.com/numba/numba/issues/6831
+        # bug in setuptools/packaging causing a deprecation warning
+        flags = ["-Werror", "-Wignore::DeprecationWarning:packaging:"]
```
Comment (bnavigator) [inline numba/tests/test_import.py:63]: I tested it, and it only works without the module or when you add `.version`.
```suggestion
        flags = ["-Werror", "-Wignore::DeprecationWarning:packaging.version:"]
```

Comment (stuartarchibald) [reply numba/tests/test_import.py]: Thanks very much, I've added the `.version` in 94fc06e, hopefully that should now work?

Comment (bnavigator) [reply numba/tests/test_import.py]: https://build.opensuse.org/package/show/home:bnavigator:branches:devel:languages:python:numeric/python-numba

Comment (stuartarchibald) [reply numba/tests/test_import.py]: Thanks for the build link, I presume https://build.opensuse.org/package/live_build_log/home:bnavigator:branches:devel:languages:python:numeric/python-numba:test/openSUSE_Tumbleweed/x86_64 means success on this particular item?

The `ppc64le` failure is https://github.com/numba/numba/issues/6832 again.
The `i586` failure is strange, `ldexp(2.5, -2)` should give 0.625 but for some reason it's giving `inf`, IIRC it's just a call through to `libm`.

Comment (bnavigator) [reply numba/tests/test_import.py]: Yes, `x86_64` shows that the import test does not fail anymore, thanks to your patch.

I think you mean `i586` is #6832.
I would not be too concerned about ppc64le. Many numeric packages don't build successfully for those platforms.

**Review by sklam — APPROVED** [review]
Thanks for the patch!

File: numba/tests/test_import.py (hunk 1 above)
Comment (stuartarchibald) [reply numba/tests/test_import.py]: > I think you mean `i586` is #6832.
> I would not be too concerned about ppc64le. Many numeric packages don't build successfully for those platforms.

Yes, apologies, got those transposed. `ppc64le` conda builds are working https://anaconda.org/numba/numba/files so I guess it must be something different about the set up.

**Pull request merged**